libmamba/include/mamba/specs/build_number_spec.hpp

```diff
@@ -124,6 +125,16 @@ namespace mamba::specs
          */
         [[nodiscard]] auto contains(BuildNumber point) const -> bool;
 
+        [[nodiscard]] auto operator==(const BuildNumberSpec& other) const -> bool
```

Comment:
Add a `// TODO: ` to replace this by  `= default;` as soon as we use C++20 for mamba, as it will help with maintenance.

Do it for every additional `operator==` in this PR, because this is introducing a potential maintenance headache (I think that's why it was not done before and also because it was not necessary).